codec-http2/src/main/java/io/netty/handler/codec/http2/Http2RemoteFlowController.java

```diff
@@ -53,6 +53,11 @@
 
         /**
          * Signal an error and release any retained buffers.
+         *<p>
+         * This method might be called multiple times.
```

Comment:
+1. Seems like we need to tighten up the error or "non-normal" completion path to ensure the lifecycle manager gets invoked.